cookbooks/aws-parallelcluster-platform/resources/dcv/partial/_ubuntu_common.rb

```diff
@@ -82,6 +82,62 @@ def optionally_disable_rnd
     end
   end
 
+  # Disable Wayland in GDM to ensure Xorg is used
+  # This is required for Ubuntu 22.04+ where Wayland is the default
+  # Without this, GDM won't start Xorg on headless GPU instances
+  def disable_wayland
+    bash 'Disable Wayland in GDM' do
+      user 'root'
+      code <<-DISABLEWAYLAND
```

Comment:
Can you add a Unit Test for this as well as the check of which server Type we are using as part of Kitchen test